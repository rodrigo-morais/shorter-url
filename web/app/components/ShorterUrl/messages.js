import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'app.components.ShorterUrlPage.header',
    defaultMessage: 'ShorterUrl',
  },
  form: {
    submit: {
      id: 'app.components.ShorterUrlPage.form.submit',
      defaultMessage: 'Convert',
    },
  },
});
