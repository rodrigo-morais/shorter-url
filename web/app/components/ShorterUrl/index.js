/* eslint no-param-reassign: ["error", { "props": false }] */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import H1 from 'components/shared/H1';
import Table from 'components/shared/Table';
import Thead from 'components/shared/Thead';
import Th from 'components/shared/Th';
import Tbody from 'components/shared/Tbody';
import Tr from 'components/shared/Tr';
import Td from 'components/shared/Td';
import Form from 'components/shared/Form';
import InputText from 'components/shared/InputText';
import Fieldset from 'components/shared/Fieldset';
import Button from 'components/shared/Button';

import messages from './messages';

const ShorterUrl = ({ urls, convertUrl }) => {
  const onSubmit = (event) => {
    event.preventDefault();

    const form = event.target;
    const url = form.querySelector('input');

    convertUrl(url.value);

    url.value = '';
  };

  return (
    <div>
      <H1>
        <FormattedMessage {...messages.header} />
      </H1>
      <Form onSubmit={onSubmit}>
        <Fieldset>
          <InputText type="text" />
          <Button type="submit">
            <FormattedMessage {...messages.form.submit} />
          </Button>
        </Fieldset>
      </Form>
      <hr />
      <Table>
        <Thead>
          <Tr>
            <Th>URL</Th>
            <Th>Short URL</Th>
          </Tr>
        </Thead>
        <Tbody>
          { urls.map((url, index) =>
            <Tr key={index}>
              <Td>
                {url.complete}
              </Td>
              <Td>
                { url.short }
              </Td>
            </Tr>
          ) }
        </Tbody>
      </Table>
    </div>
  );
};

ShorterUrl.propTypes = {
  urls: PropTypes.array,
  convertUrl: PropTypes.func.isRequired,
};

export default ShorterUrl;
