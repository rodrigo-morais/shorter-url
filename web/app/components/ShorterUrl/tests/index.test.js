import React from 'react';
import { IntlProvider, FormattedMessage } from 'react-intl';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ShorterUrl from '../index';
import messages from '../messages';


describe('<ShorterUrl />', () => {
  const urls = [
    { complete: 'http://www.google.com',
      short: 'http:/goo.gl',
    },
  ];

  it('renders according to design', () => {
    const component = renderer.create(
      <IntlProvider locale="en">
        <ShorterUrl
          urls={urls}
          loading={false}
          error={false}
        />
      </IntlProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the page title', () => {
    const renderedComponent = shallow(
      <ShorterUrl
        urls={urls}
        loading={false}
        error={false}
      />
    );
    expect(renderedComponent.contains(
      <FormattedMessage {...messages.header} />
    )).toEqual(true);
  });
});
