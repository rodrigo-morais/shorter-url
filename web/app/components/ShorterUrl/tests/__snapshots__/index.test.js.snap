exports[`<ShorterUrl /> renders according to design 1`] = `
<div>
  <h1
    className="ccoFJN">
    <span>
      ShorterUrl
    </span>
  </h1>
  <form
    className="eQdiUA"
    onSubmit={[Function]}>
    <fieldset
      className="gDzNqM">
      <input
        className="mBQoT"
        type="text" />
      <button
        className="fkphIJ"
        type="submit">
        <span>
          Convert
        </span>
      </button>
    </fieldset>
  </form>
  <hr />
  <table
    className="cmoJIt">
    <thead
      className="foqgVb">
      <tr
        className="foqgVb">
        <th
          className="dihJkC">
          URL
        </th>
        <th
          className="dihJkC">
          Short URL
        </th>
      </tr>
    </thead>
    <tbody
      className="foqgVb">
      <tr
        className="foqgVb">
        <td
          className="ePQBtB">
          http://www.google.com
        </td>
        <td
          className="ePQBtB">
          http:/goo.gl
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
